lib/puppet/util/windows/adsi.rb

```diff
@@ -491,14 +488,14 @@ def self.create(name)
     def self.exists?(name_or_sid)
       well_known = false
       if (sid = Puppet::Util::Windows::SID.name_to_sid_object(name_or_sid))
-        return true if sid.account_type == 'group'
+        return true if sid[:type] == :SidTypeGroup
 
         # 'well known group' is special as it can be a group like Everyone OR a user like SYSTEM
         # so try to resolve it
         # https://msdn.microsoft.com/en-us/library/cc234477.aspx
-        well_known = sid.account_type == 'well known group'
-        return false if sid.account_type != 'alias' && !well_known
-        name_or_sid = "#{sid.domain}\\#{sid.account}"
+        well_known = sid[:type] == :SidTypeWellKnownGroup
```

Comment:
No, Ruby hashes don't work like that (I think you might be thinking of PowerShell).

```
irb(main):001:0> foo = { :bar => 2, :baz => 'hello' }
=> {:bar=>2, :baz=>"hello"}
irb(main):002:0> foo[:bar]
=> 2
irb(main):003:0> foo.bar
NoMethodError: undefined method `bar' for {:bar=>2, :baz=>"hello"}:Hash
        from (irb):3
        from C:/tools/ruby215/bin/irb:11:in `<main>'
irb(main):004:0>
```